android/src/main/java/com/expensify/livemarkdown/RangeSplitter.java

```diff
@@ -41,13 +44,16 @@ public static ArrayList<MarkdownRange> splitRangesOnEmojis(@NonNull List<Markdow
           MarkdownRange newRange = new MarkdownRange(currentRange.getType(), currentStart, emojiStart - currentStart, currentRange.getDepth());
           currentRange = new MarkdownRange(currentRange.getType(), emojiEnd, currentEnd - emojiEnd, currentRange.getDepth());
 
-          newRanges.add(newRange);
+          if (newRange.getLength() > 0) {
+           newRanges.add(newRange);
+          }
```

Comment:
Shouldn't this check be symmetric? So it covers both `_text😀_` and `_😀text_`?

Also, let's rename `newRange` and `currentRange` to `leftRange` and `rightRange`.

Let's avoid creating `newRange` object at all if possible, i.e. move the constructor call into the conditional.